### PR TITLE
feat(api-idorslug): Updated Subset of Orgmember and Integration Endpoints to use `organization_id_or_slug`

### PR DESCRIPTION
**Pull request by iamrajjoshi**
A subset of changes from https://github.com/getsentry/sentry/pull/70081!

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -120,14 +120,12 @@ ignore_missing_imports = true
 module = [
     "sentry.api.base",
```
Comment (iamrajjoshi) [inline pyproject.toml:121]: File updated because I improved typing

File: src/sentry/api/bases/organization.py [hunk 2]
```diff
@@ -242,30 +242,42 @@ class ControlSiloOrganizationEndpoint(Endpoint):
     def convert_args(
         self,
         request: Request,
-        organization_slug: str | int | None = None,
         *args: Any,
         **kwargs: Any,
     ) -> tuple[tuple[Any, ...], dict[str, Any]]:
+        organization_id_or_slug: int | str | None = None
```
Comment (iamrajjoshi) [inline src/sentry/api/bases/organization.py:248]: This is a temporary jerry rig so that we maintain backwards compatibility with getsentry endpoints and other endpoints which i haven't merged yet until I update them. They expect convert_args to accept organization_slug, so I can't keep it named, I have to keep it unnamed and extract it. I don't like it, but couldn't think of a better way.

File: src/sentry/api/bases/organization.py [hunk 3]
```diff
@@ -242,30 +242,42 @@ class ControlSiloOrganizationEndpoint(Endpoint):
...
+        if args and args[0] is not None:
```
Comment (ykamo001) [inline src/sentry/api/bases/organization.py:249]: but you're allowing `organization_id_or_slug` to be None right, so does it make sense still to check for not none?

Comment (iamrajjoshi) [reply src/sentry/api/bases/organization.py:249]: hmm, i don't think that check does anything. if no function call will pass none to the method because the serializer will complain if nothing is passed as the `org_id_or_slug` to the endpoint.

Comment (ykamo001) [reply src/sentry/api/bases/organization.py:249]: > hmm, i don't think that check does anything. if no function call will pass none to the method because the serializer will complain if nothing is passed as the `org_id_or_slug` to the endpoint.

I don't think I follow sorry, could you rephrase or explain with example?

Comment (iamrajjoshi) [reply src/sentry/api/bases/organization.py:249]: What i meant is args[0] will never be None, so really that check doesn't do anything, Ill delete it.

File: src/sentry/api/bases/organization.py [hunk 4]
```diff
@@ -242,30 +242,42 @@ class ControlSiloOrganizationEndpoint(Endpoint):
...
+            args = args[1:]
+        else:
+            organization_id_or_slug = kwargs.pop("organization_id_or_slug", None) or kwargs.pop(
+                "organization_slug", None
+            )
+
         if not subdomain_is_region(request):
             subdomain = getattr(request, "subdomain", None)
-            if subdomain is not None and subdomain != organization_slug:
+            if subdomain is not None and subdomain != organization_id_or_slug:
```
Comment (ykamo001) [inline src/sentry/api/bases/organization.py:261]: `organization_id_or_slug` can be `None, so are we okay with this evaluating into:
```
subdomain != None:
```
which will always be true because of the first check at `subdomain is not None` 😆

Comment (iamrajjoshi) [reply src/sentry/api/bases/organization.py:261]: yeah the types are a mess here, but if `organization_id_or_slug` is None, we will raise ResourceDoesNotExist because we do this check right under here. So this logic here is generally only for when its not None. I should probably move the conditional above this.

File: src/sentry/api/bases/organization.py [hunk 5]
```diff
@@ -536,32 +548,42 @@ def get_filter_params(
     def convert_args(
         self,
         request: Request,
-        organization_slug: str | int | None = None,
         *args: Any,
         **kwargs: Any,
     ) -> tuple[tuple[Any, ...], dict[str, Any]]:
         """
-        We temporarily allow the organization_slug to be an integer as it actually can be both slug or id
+        We temporarily allow the organization_id_or_slug to be an integer as it actually can be both slug or id
```
Comment (sentaur-athena) [inline src/sentry/api/bases/organization.py:555]: nit: I don't think this comment is valid anymore.

File: src/sentry/api/endpoints/codeowners/external_actor/user_details.py [hunk 6]
```diff
@@ -27,7 +27,7 @@ class ExternalUserDetailsEndpoint(OrganizationEndpoint, ExternalActorEndpointMix
     }
     owner = ApiOwner.ENTERPRISE
 
-    def convert_args(  # type: ignore[override]
```
Comment (sentaur-athena) [inline src/sentry/api/endpoints/codeowners/external_actor/user_details.py:30]: I don't see any type fixes here :/ how come you could remove the comment? 😅

**Review by sentaur-athena — APPROVED** [review]

File: src/sentry/api/endpoints/codeowners/external_actor/user_details.py (hunk 6 above)
Comment (iamrajjoshi) [reply src/sentry/api/endpoints/codeowners/external_actor/user_details.py:30]: The fix was upstream. Before, we use to two a mismatched function signature with the superclass

**Review by ykamo001 — APPROVED** [review]
